### ISSUE TITLE
Crash/panic because of "unexpected fault address"

### ISSUE THREAD

**Issue by mrbuk** [cloudfoundry-community/firehose-to-syslog#106]
i do get from time to time the following panic

```
[2016-07-08 16:17:13.775896206 +0200 CEST] Firehose Subscription Succesfull! Routing events...
unexpected fault address 0x7f3bae94e500
fatal error: fault
[signal 0xb code=0x1 addr=0x7f3bae94e500 pc=0x5b610d]

goroutine 1 [running]:
runtime.throw(0x9a67a8, 0x5)
        /usr/local/go/src/runtime/panic.go:547 +0x90 fp=0xc82017f638 sp=0xc82017f620
runtime.sigpanic()
        /usr/local/go/src/runtime/sigpanic_unix.go:27 +0x2ab fp=0xc82017f688 sp=0xc82017f638
encoding/json.checkValid(0x7f3bae94e500, 0xe6, 0x7ffffb4f, 0xc8200e5a60, 0x0, 0x0)
        /usr/local/go/src/encoding/json/scanner.go:22 +0xad fp=0xc82017f6e0 sp=0xc82017f688
encoding/json.Unmarshal(0x7f3bae94e500, 0xe6, 0x7ffffb4f, 0x7f5400, 0xc82026c480, 0x0, 0x0)
        /usr/local/go/src/encoding/json/decode.go:90 +0x73 fp=0xc82017f750 sp=0xc82017f6e0
github.com/cloudfoundry-community/firehose-to-syslog/vendor/github.com/pquerna/ffjson/ffjson.Unmarshal(0x7f3bae94e500, 0xe6, 0x7ffffb4f, 0x7f5400, 0xc82026
c480, 0x0, 0x0)
        /go/src/github.com/cloudfoundry-community/firehose-to-syslog/vendor/github.com/pquerna/ffjson/ffjson/marshal.go:95 +0x3d2 fp=0xc82017f838 sp=0xc820
17f750
github.com/cloudfoundry-community/firehose-to-syslog/caching.GetAppInfo(0xc8202bef30, 0x24, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
        /go/src/github.com/cloudfoundry-community/firehose-to-syslog/caching/caching.go:109 +0x13f fp=0xc82017f938 sp=0xc82017f838
github.com/cloudfoundry-community/firehose-to-syslog/events.getAppInfo(0xc8202bef30, 0x24, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
        /go/src/github.com/cloudfoundry-community/firehose-to-syslog/events/events.go:121 +0x54 fp=0xc82017fa70 sp=0xc82017f938
github.com/cloudfoundry-community/firehose-to-syslog/events.(*Event).AnnotateWithAppData(0xc82017fc90)
        /go/src/github.com/cloudfoundry-community/firehose-to-syslog/events/events.go:301 +0x23c fp=0xc82017fc30 sp=0xc82017fa70
github.com/cloudfoundry-community/firehose-to-syslog/events.routeEvent(0xc820086480, 0xc8204e2090)
        /go/src/github.com/cloudfoundry-community/firehose-to-syslog/events/events.go:61 +0x131 fp=0xc82017fcc0 sp=0xc82017fc30
github.com/cloudfoundry-community/firehose-to-syslog/events.RouteEvents(0xc8201ae0c0, 0xc8204e2090)
        /go/src/github.com/cloudfoundry-community/firehose-to-syslog/events/events.go:28 +0x6e fp=0xc82017fcf8 sp=0xc82017fcc0
main.main()
        /go/src/github.com/cloudfoundry-community/firehose-to-syslog/main.go:130 +0xee7 fp=0xc82017ff50 sp=0xc82017fcf8
runtime.main()
        /usr/local/go/src/runtime/proc.go:188 +0x2b0 fp=0xc82017ffa0 sp=0xc82017ff50
runtime.goexit()
        /usr/local/go/src/runtime/asm_amd64.s:1998 +0x1 fp=0xc82017ffa8 sp=0xc82017ffa0

goroutine 17 [syscall, 1566 minutes, locked to thread]:
runtime.goexit()
        /usr/local/go/src/runtime/asm_amd64.s:1998 +0x1

goroutine 5 [syscall, 1566 minutes]:
os/signal.signal_recv(0x0)
        /usr/local/go/src/runtime/sigqueue.go:116 +0x132
os/signal.loop()
        /usr/local/go/src/os/signal/signal_unix.go:22 +0x18
created by os/signal.init.1
        /usr/local/go/src/os/signal/signal_unix.go:28 +0x37

goroutine 10 [chan send]:
github.com/cloudfoundry-community/firehose-to-syslog/vendor/github.com/cloudfoundry/noaa/consumer.(*Consumer).firehose.func1(0xc820086510)
        /go/src/github.com/cloudfoundry-community/firehose-to-syslog/vendor/github.com/cloudfoundry/noaa/consumer/async.go:171 +0x40
github.com/cloudfoundry-community/firehose-to-syslog/vendor/github.com/cloudfoundry/noaa/consumer.(*Consumer).listenForMessages(0xc8200cc240, 0xc8205ea080,
 0xc820013240, 0x0, 0x0)
        /go/src/github.com/cloudfoundry-community/firehose-to-syslog/vendor/github.com/cloudfoundry/noaa/consumer/async.go:211 +0x2d7
github.com/cloudfoundry-community/firehose-to-syslog/vendor/github.com/cloudfoundry/noaa/consumer.(*Consumer).listenAction.func1(0x0, 0x0, 0xc8200cc240)
        /go/src/github.com/cloudfoundry-community/firehose-to-syslog/vendor/github.com/cloudfoundry/noaa/consumer/async.go:225 +0x15b
github.com/cloudfoundry-community/firehose-to-syslog/vendor/github.com/cloudfoundry/noaa/consumer.(*Consumer).retryAction(0xc8200cc240, 0xc8206460c0, 0xc82
01ae120, 0x5)
        /go/src/github.com/cloudfoundry-community/firehose-to-syslog/vendor/github.com/cloudfoundry/noaa/consumer/async.go:243 +0x10d
github.com/cloudfoundry-community/firehose-to-syslog/vendor/github.com/cloudfoundry/noaa/consumer.(*Consumer).firehose.func2(0xc8201ae120, 0xc8201ae0c0, 0x
c8200cc240, 0xc8205ea080, 0xc82041e0f0, 0x21, 0xc820246f00, 0x4da, 0xc820013240, 0x5)
        /go/src/github.com/cloudfoundry-community/firehose-to-syslog/vendor/github.com/cloudfoundry/noaa/consumer/async.go:179 +0xec
created by github.com/cloudfoundry-community/firehose-to-syslog/vendor/github.com/cloudfoundry/noaa/consumer.(*Consumer).firehose
        /go/src/github.com/cloudfoundry-community/firehose-to-syslog/vendor/github.com/cloudfoundry/noaa/consumer/async.go:180 +0x199

```

I am running tag/release 1.6.0 which I have compiled myself with Go 1.6.2 (see https://gist.github.com/mrbuk/8b0bb7a34391c8f67ce90c23db631b00) due to no available `musl` library in CentOS 6.

Maybe its somehow related to 

> Using references to keys or values outside of the transaction they were obtained from can cause them to show wrong data or cause an unexpected fault address panic (via #204 / @calmh).

https://github.com/boltdb/bolt/issues/203

Maybe [unmarshalling](https://github.com/cloudfoundry-community/firehose-to-syslog/blob/1.6.0/caching/caching.go#L109) should be moved into the [bolt transaction block](https://github.com/cloudfoundry-community/firehose-to-syslog/blob/1.6.0/caching/caching.go#L106)?

**Comment by shinji62** [comment]
I will check thanks.

**Comment by shinji62** [comment]
Should be fixed in 1.7.2 release.
Please reopen if you still see some panic

**Comment by mrbuk** [comment]
Thanks a lot. I will keep to posted.
